pkg/users/account-create-dialog.js

```diff
@@ -276,15 +366,21 @@
                             "-e",
                             user_name
                         ], { superuser: "require", err: "message" });
+                })
+                .then(() => {
+                    if (force_home) {
+                        return cockpit.spawn(["id", user_name, "--group"], { superuser: "require", err: "message" })
+                                .then(gid => cockpit.spawn(["chown", "-hR", `${user_name}:${gid.trim()}`, home_dir], { superuser: "require", err: "message" }));
+                    }
                 });
     }
 
-    function passwd_check(force_weak, real_name, user_name, password, password_confirm, locked, force_change) {
-        return validate(force_weak, real_name, user_name, password, password_confirm).then(valid => {
+    function passwd_check(force_weak, force_home, real_name, user_name, password, password_confirm, locked, home_dir, force_change) {
+        return validate(force_weak, force_home, real_name, user_name, password, password_confirm).then(valid => {
             if (valid)
-                return create(real_name, user_name, password, locked, force_change);
+                return create(real_name, user_name, password, locked, force_change, home_dir, force_home);
             else {
-                if (!errors.real_name && !errors.user_name && !errors.password_confirm && state.password.length <= 256) {
+                if (!errors.real_name && !errors.user_name && !errors.home_dir && !errors.password_confirm && state.password.length <= 256) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18234-20230317-103941-df531fdc-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)